### PR TITLE
feat: introduce default interactive mode

### PR DESCRIPTION
**Pull request by slandelle**
Motivation:

We want to align the maven and the gradle plugins behavior.

In particular, we want it to be interactive by default, with a simulation selection prompt.

### CODE REVIEW

File: src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy [hunk 1]
```diff
@@ -17,16 +17,13 @@ final class GatlingPlugin implements Plugin<Project> {
 
     public static def GATLING_RECORDER_TASK_NAME = 'gatlingRecorder'
 
-    static String GATLING_TASK_NAME_PREFIX = "$GATLING_RUN_TASK_NAME-"
```
Comment (slandelle) [inline src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy:20]: `gatlingRun-FQCN` is dropped in favor of `gatling --simulation=FQCN`

File: src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy [hunk 2]
```diff
@@ -35,6 +32,8 @@ final class GatlingPlugin implements Plugin<Project> {
 
         GatlingPluginExtension gatlingExt = project.extensions.create(GATLING_EXTENSION_NAME, GatlingPluginExtension)
 
+        validateGatlingVersion(gatlingExt.gatlingVersion)
```
Comment (slandelle) [inline src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy:35]: This release will require at least Gatling 3.11.0.

File: src/test/groovy/func/GatlingRunTaskTest.groovy [hunk 3]
```diff
@@ -22,7 +22,7 @@ class GatlingRunTaskTest extends GatlingFuncSpec {
 
     def "should resolve simulations using default filter"() {
         when:
-        def gatlingRunSimulations = theTask.simulationClasses()
+        def gatlingRunSimulations = theTask.simulationClasses(false)
```
Comment (Isammoc) [inline src/test/groovy/func/GatlingRunTaskTest.groovy:25]: please remove this unused `boolean`s

**Review by Isammoc — APPROVED** [review]

**Review by guilgaly — APPROVED** [review]

File: src/test/resources/gradle-layout-kotlin/src/gatling/resources/search.csv [hunk 4]
```diff
@@ -0,0 +1,3 @@
+searchCriterion,searchComputerName
+Macbook,MacBook Pro
+eee,ASUS Eee PC 1005PE
```
Comment (Isammoc) [inline src/test/resources/gradle-layout-kotlin/src/gatling/resources/search.csv:3]: EOL

File: src/test/resources/gradle-layout-kotlin/src/main/java/computerdatabase/MainUtils.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package computerdatabase;
+
+public class MainUtils {
+    public static String hostName() {
+        return System.getProperty("gatling.hostName");
+    }
+}
```
Comment (Isammoc) [inline src/test/resources/gradle-layout-kotlin/src/main/java/computerdatabase/MainUtils.java:7]: EOL

File: src/test/resources/gradle-layout-kotlin/src/test/java/computerdatabase/TestUtils.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package computerdatabase;
+
+public class TestUtils {
+    public static String hostName() {
+        return MainUtils.hostName();
+    }
+}
```
Comment (Isammoc) [inline src/test/resources/gradle-layout-kotlin/src/test/java/computerdatabase/TestUtils.java:7]: EOL

File: src/test/resources/gradle-layout-scala/src/gatling/resources/search.csv [hunk 7]
```diff
@@ -0,0 +1,3 @@
+searchCriterion,searchComputerName
+Macbook,MacBook Pro
+eee,ASUS Eee PC 1005PE
```
Comment (Isammoc) [inline src/test/resources/gradle-layout-scala/src/gatling/resources/search.csv:3]: EOL

File: src/test/resources/gradle-layout-scala/src/main/java/computerdatabase/MainUtils.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package computerdatabase;
+
+public class MainUtils {
+    public static String hostName() {
+        return System.getProperty("gatling.hostName");
+    }
+}
```
Comment (Isammoc) [inline src/test/resources/gradle-layout-scala/src/main/java/computerdatabase/MainUtils.java:7]: EOL

File: src/test/resources/gradle-layout-scala/src/test/java/computerdatabase/TestUtils.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
+package computerdatabase;
+
+public class TestUtils {
+    public static String hostName() {
+        return MainUtils.hostName();
+    }
+}
```
Comment (Isammoc) [inline src/test/resources/gradle-layout-scala/src/test/java/computerdatabase/TestUtils.java:7]: EOL

**Review by Isammoc — APPROVED** [review]
nits

Wonderful :clap:

File: src/test/resources/gradle-layout-kotlin/src/gatling/resources/search.csv (hunk 4 above)
Comment (slandelle) [reply src/test/resources/gradle-layout-kotlin/src/gatling/resources/search.csv:3]: I have to check what our CSV library does on a last empty line. We don't want an empty record.

**Pull request merged**